glue/viewers/image/viewer.py

```diff
@@ -214,24 +214,26 @@ def _script_header(self):
 
         script = ""
         script += "fig, ax = init_mpl(wcs=True)\n"
-        script += "ax.set_aspect('{0}')\n".format(self.state.aspect)
+        script += f"ax.set_aspect('{self.state.aspect}')\n"
 
         script += '\ncomposite = CompositeArray()\n'
-        script += "image = imshow(ax, composite, origin='lower', interpolation='nearest', aspect='{0}')\n\n".format(self.state.aspect)
+        script += f"image = imshow(ax, composite, origin='lower', interpolation='nearest', aspect='{self.state.aspect}')\n\n"
 
         dindex = self.session.data_collection.index(self.state.reference_data)
 
-        script += "ref_data = data_collection[{0}]\n".format(dindex)
+        script += f"ref_data = data_collection[{dindex}]\n"
 
         ref_coords = self.state.reference_data.coords
 
-        if hasattr(ref_coords, 'wcs'):
-            script += "ax.reset_wcs(slices={0}, wcs=ref_data.coords.wcs)\n".format(self.state.wcsaxes_slice)
+        if isinstance(ref_coords, WCS):
+            script += f"ax.reset_wcs(slices={self.state.wcsaxes_slice}, wcs=ref_data.coords)\n"
```

Comment:
In principle we should be able to pass any ``Coordinates`` subclass to ``reset_wcs`` - after all this is what we do in the actual image viewer:

```
        ref_coords = getattr(self.state.reference_data, 'coords', None)

        if ref_coords is None or isinstance(ref_coords, LegacyCoordinates):
            self.axes.reset_wcs(slices=self.state.wcsaxes_slice,
                                wcs=get_identity_wcs(self.state.reference_data.ndim))
        else:
            self.axes.reset_wcs(slices=self.state.wcsaxes_slice, wcs=ref_coords)
```

If we don't care about legacy coordinates in the script export, we might be able to get away with just doing:

```
self.axes.reset_wcs(slices=self.state.wcsaxes_slice, wcs=ref_coords)
```

if ``ref_coords`` is defined? (and not have the case below either)

We could test that this works using e.g. ``AffineCoordinates``.

Reply:
> If we don't care about legacy coordinates in the script export, we might be able to get away with just doing:

Basically those are the `else` case already covered, but if all the rest can be treated the same way, it becomes much easier of course (I haven't seen an example using `coords.wcs` or `coords.wcs_dict`).
No idea if those `AffineCoordinates` make any sense at all, but they work – apart from the `_slice` and `simple_att` tests, that suffer from some fuzziness in the (lower axis) labels. Can probably live with that?
![glue_plot](https://user-images.githubusercontent.com/709020/195990101-7af8a541-7bdb-4065-a110-ff1f1e472211.png)
![expected](https://user-images.githubusercontent.com/709020/195990103-ecc38c94-8d25-4d04-8f86-54bec0d7360f.png)